src/Sweep.cpp

```diff
@@ -291,131 +291,246 @@ void InfectSweep(double t, int run) //added run number as argument in order to r
 	double fp; //// false positive
 	unsigned short int ts;
 
-	if (!P.DoSeasonality)	seasonality = 1.0;
-	else					seasonality = P.Seasonality[((int)t) % DAYS_PER_YEAR];
+	// if not doing seasonality
+	if (!P.DoSeasonality)
+	{	
+		// set seasonality to 1
+		seasonality = 1.0;
+	}
+	else{
+		// otherwise pick seasonality from P.Seasonality array using day number in year
```

Comment:
I'd like to see this refactored into a function with that sort of name (ie `pick_seasonality_from_day_number(int day_number)`), rather than described in a comment.

Reply:
Absolutely agree. There is ample scope for refactoring.